File: tundra-cli/benchmarks/sieve/sieve.cpp
// File: sieve/sieve.cpp
#include <iostream>
#include <vector>
#include <cstdlib>

int main(int argc, char** argv) {
    int n = 200000;
    if (argc > 1) {
        n = std::atoi(argv[1]);
    }
    std::vector<bool> isComposite(n + 1, false);
    for (int p = 2; p * p <= n; ++p) {
        if (!isComposite[p]) {
            for (int m = p * p; m <= n; m += p) {
                isComposite[m] = true;
            }
        }
    }
    int count = 0;
    for (int i = 2; i <= n; ++i) {
        if (!isComposite[i]) count++;
    }
    std::cout << count << "\n";
    return 0;
}

File: tundra-cli/benchmarks/sieve/sieve.go
// File: sieve/sieve.go
package main

import (
    "fmt"
    "os"
    "strconv"
)

func sieve(n int) int {
    isComposite := make([]bool, n+1)
    for p := 2; p*p <= n; p++ {
        if !isComposite[p] {
            for m := p * p; m <= n; m += p {
                isComposite[m] = true
            }
        }
    }
    count := 0
    for i := 2; i <= n; i++ {
        if !isComposite[i] {
            count++
        }
    }
    return count
}

func main() {
    n := 200000
    if len(os.Args) > 1 {
        if v, err := strconv.Atoi(os.Args[1]); err == nil {
            n = v
        }
    }
    fmt.Println(sieve(n))
}
